archivessnake/edit_notes.py

```diff
@@ -7,64 +7,67 @@
 import time
 
 from asnake.aspace import ASpace
+from asnake.utils import walk_tree
 #from fuzzywuzzy import fuzz
 from rapidfuzz import fuzz
 
-AS_REPOSITORY = 2
 OUTPUT_FILENAME = "out.csv"
 NOTE_TYPE_CHOICES = ["bioghist", "accessrestrict", "odd", "abstract", "arrangement", "userestrict", "fileplan", "acqinfo", "langmaterial", "physdesc", "prefercite", "processinfo", "relatedmaterial", "separatedmaterial"]
 ACTION_CHOICES = ["modify", "delete"]
 LEVEL = ["collection", "file", "series", "item", "all"]
 CONFIDENCE_RATIO = 97
-spreadsheet_path = os.path.join(os.path.abspath(os.path.dirname(__file__)), OUTPUT_FILENAME)
 
-def process_tree(args, resource):
+def process_tree(client, parser, writer):
     """Iterates through a given collection, file, or series provided by user input. Finds note content that matches user input and then deletes or modifies relevant notes according to user preference.  """
-    for record in resource.tree.walk:
+    args = parser.parse_args()
```

Comment:
I know this allows you to pass fewer arguments, but I think it's a bit of an anti-pattern in that there's no clear relationship between the arguments you're creating in `get_parser` and then what's being used throughout the rest of the script. I would prefer to see something slightly different:

- set and parse the args as the _first_ thing in the script. You can do this in the first few lines of `main` OR even in the block below (lines 96). That way, if someone doesn't pass the expected arguments, the script doesn't do anything. As it is, it will try and execute a bunch of stuff before actually parsing the arguments.
- Pass ALL the arguments you need into `process_tree`. This will mean you have a lot of arguments, but that's okay!
- Pass just the arguments you need into other methods (like `handle_matching_notes` etc)

That approach makes clear what's being used where, and also is much easier to write unit tests against.